### PR TITLE
fix: fix makefiles

### PR DESCRIPTION
**Pull request by Ratakor**
## Notice
- [x] I *understand* the code that I have edited, and have the means
to test it before making changes to Concord.

## What?
(commit message)
Clean some whitespaces.
Remove -e for echo because it is non-standard (bmake).
Add 'all' dependency for main Makefile's 'install' so it is possible to directly do `make install` instead of `make all install`.
Fix clean so it doesn't delete gencodecs objects and fix purge so it 'clean' and call 'clean' for gencodecs as I assume it was how it supposed to work.

## Why?
Because "make clean install" is less to type than "make clean all install" and gencodecs takes a long time to compile.

### CODE REVIEW

**Comment by HackerSmacker** [comment]
Honestly, we should probably trim whitespace from every file in the project, I'd be willing to bet that there's a fair bit of it all over the place. 

As for the `make clean` fix, I think it's okay, but, something I've always watched out for is accidentally "shipping" a preprocessed/generated discord-codecs.c, but, that's what .gitignore files are for. 

I legitimately do not think anyone has ever used `make echo` before. Even though it does use the POSIX.1-2002 `echo -e`, it still "works" on systems with old echo, albeit at the expense of seeing that -e in the output. 

I'm voting yes on this.

File: core/Makefile [hunk 1]
```diff
@@ -22,11 +22,11 @@ CFLAGS += -std=c99 -pthread -D_XOPEN_SOURCE=600 -DLOG_USE_COLOR \
 all: $(OBJS)
 
 echo:
-	@ echo -e 'CC: $(CC)\n'
-	@ echo -e 'CFLAGS: $(CFLAGS)\n'
-	@ echo -e 'OBJS: $(OBJS)\n'
+	@ echo 'CC: $(CC)'
+	@ echo 'CFLAGS: $(CFLAGS)'
+	@ echo 'OBJS: $(OBJS)'
```
Comment (HackerSmacker) [inline core/Makefile:27]: Honestly, we could probably just remove `make echo`.

**Review by HackerSmacker — APPROVED** [review]
Apparently I forgot to file an official review, so, here we go!

**Comment by jdeokkim** [comment]
Wow!

**Comment by Ratakor** [comment]
> Wow!

uh ?

**Pull request closed without merging**